test/e2e/copy_from_image_test.go

```diff
@@ -126,31 +126,63 @@ func TestCopyImageInputToTarWithNonDistributableLayersFlagButContainsANonDistrib
 }
 
 func TestCopyAnImageFromATarToARepoThatDoesNotContainNonDistributableLayersButTheFlagWasIncluded(t *testing.T) {
-	env := BuildEnv(t)
-	imgpkg := Imgpkg{t, Logger{}, env.ImgpkgPath}
-	defer env.Cleanup()
+	t.Run("environment with internet", func(t *testing.T) {
+		env := BuildEnv(t)
+		imgpkg := Imgpkg{t, Logger{}, env.ImgpkgPath}
+		defer env.Cleanup()
 
-	// general setup
-	testDir := env.Assets.CreateTempFolder("image-to-tar")
-	tarFilePath := filepath.Join(testDir, "image.tar")
+		// general setup
+		testDir := env.Assets.CreateTempFolder("image-to-tar")
+		tarFilePath := filepath.Join(testDir, "image.tar")
 
-	env.ImageFactory.PushImageWithANonDistributableLayer(env.RelocationRepo)
+		env.ImageFactory.PushImageWithANonDistributableLayer(env.RelocationRepo)
 
-	repoToCopyName := env.RelocationRepo + "include-non-distributable"
-	var stdOutWriter bytes.Buffer
+		repoToCopyName := env.RelocationRepo + "include-non-distributable"
+		var stdOutWriter bytes.Buffer
 
-	// copy to tar skipping NDL
-	imgpkg.Run([]string{"copy", "-i", env.RelocationRepo, "--to-tar", tarFilePath})
+		// copy to tar skipping NDL
+		imgpkg.Run([]string{"copy", "-i", env.RelocationRepo, "--to-tar", tarFilePath})
 
-	imgpkg.RunWithOpts([]string{"copy", "--tar", tarFilePath, "--to-repo", repoToCopyName, "--include-non-distributable"}, RunOpts{
-		AllowError:   true,
-		StderrWriter: &stdOutWriter,
-		StdoutWriter: &stdOutWriter,
+		imgpkg.RunWithOpts([]string{"copy", "--tar", tarFilePath, "--to-repo", repoToCopyName, "--include-non-distributable"}, RunOpts{
+			AllowError:   true,
```

Comment:
is Allowing Error here intentional? if so what error are we expecting? are there other assertions besides hint message we should be doing?

Reply:
It is intentional, the error we are expecting is on the next line which is that we expect a message explaining that the non-distributable layers could not be copied from the tarball since they were missing. There is a hint that the initial copy was missing the `--include-non-distributable` flag (hence why the non-distributable layers are missing).